ethcontract/src/transaction/gas_price.rs

```diff
@@ -1,121 +1,60 @@
 //! Implementation of gas price estimation.
 
-use crate::errors::ExecutionError;
 use primitive_types::U256;
-use web3::api::Web3;
-use web3::Transport;
+use web3::types::U64;
 
 /// The gas price setting to use.
 #[derive(Clone, Copy, Debug, PartialEq)]
 pub enum GasPrice {
-    /// The standard estimated gas price from the node, this is usually the
-    /// median gas price from the last few blocks. This is the default gas price
-    /// used by transactions.
-    Standard,
-    /// A factor of the estimated gas price from the node. `GasPrice::Standard`
-    /// is similar to `GasPrice::Scaled(1.0)` but because of how the scaling is
-    /// calculated, `GasPrice::Scaled(1.0)` can lead to some rounding errors
-    /// caused by converting the estimated gas price from the node to a `f64`
-    /// and back.
-    Scaled(f64),
-    /// Specify a specific gas price to use for the transaction. This will cause
-    /// the transaction `SendFuture` to not query the node for a gas price
-    /// estimation.
-    Value(U256),
+    /// Legacy type of transactions, using single gas price value. Equivalent to sending
+    /// eip1559 transaction with max_fee_per_gas = max_priority_fee_per_gas = gas_price
+    Legacy(U256),
+
+    /// Eip1559 type of transactions, using two values (max_fee_per_gas, max_priority_fee_per_gas)
+    Eip1559((U256, U256)),
 }
 
 impl GasPrice {
-    /// A low gas price. Using this may result in long confirmation times for
-    /// transactions, or the transactions not being mined at all.
-    pub fn low() -> Self {
-        GasPrice::Scaled(0.8)
-    }
-
-    /// A high gas price that usually results in faster mining times.
-    /// transactions, or the transactions not being mined at all.
-    pub fn high() -> Self {
-        GasPrice::Scaled(6.0)
-    }
-
-    /// Returns `Some(value)` if the gas price is explicitly specified, `None`
-    /// otherwise.
-    pub fn value(&self) -> Option<U256> {
+    /// Prepares the data for transaction. Returns tuple:
+    /// (gas_price, max_fee_per_gas, max_priority_fee_per_gas, transaction_type)
+    pub fn resolve_for_transaction(
+        &self,
+    ) -> (Option<U256>, Option<U256>, Option<U256>, Option<U64>) {
```

Comment:
Might make sense define a structure for this return value? In general (like :point_up:) I find that these larger tuples with where many fields have the same type make it easy to accidentally get the order of the return values hard, which makes for some hard to find bugs and some extra burden on reviews:

```rust
struct ResolvedTransactionGasPrice {
  pub gas_price: Option<U256>,
  pub max_fee_per_gas: Option<U256>,
  pub max_priority_fee_per_gas: Option<U256>,
  pub transaction_type: Option<U64>,
}
```

Reply:
Agreed. Fixed.